File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.py
from typing import List

class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        index_stack = []
        output = [0] * len(temperatures)

        for i in range(len(temperatures)):
            while index_stack and temperatures[index_stack[-1]] < temperatures[i]:
                popped_index = index_stack.pop()
                output[popped_index] = i - popped_index

            index_stack.append(i)

        return output

File: LeetCode/Medium/0739-daily-temperatures/0739-daily-temperatures.java
import java.util.Stack;

class Solution {
        public int[] dailyTemperatures(int[] temperatures) {
            Stack<Integer> indexStack = new Stack<>();
            int[] result = new int[temperatures.length];
            
            for (int i = 0; i < temperatures.length; i++) {
                
                while(!indexStack.isEmpty() && temperatures[indexStack.peek()] < temperatures[i]) {
                    
                    result[indexStack.peek()] = i - indexStack.pop();
                }
                
                indexStack.push(i);
            }
            
            return result;
    }
}